android/src/main/java/com/reactnativecommunity/viewpager/VerticalViewPager.java

```diff
@@ -0,0 +1,117 @@
+package com.reactnativecommunity.viewpager;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+
+import androidx.viewpager.widget.ViewPager;
+
+// Vertical ViewPager implement, original code from
+// https://android.googlesource.com/platform/packages/apps/DeskClock/+/master/src/com/android/deskclock/VerticalViewPager.java
+public class VerticalViewPager extends ViewPager {
+    private boolean mVertical = false;
+    private GestureDetector mGestureDetector;
+
+    public VerticalViewPager(Context context) {
+        super(context);
+    }
+
+    public void setOrientation(boolean vertical) {
+        mVertical = vertical;
+        if (!mVertical) return;
```

Comment:
The thing with this is that if we change the prop dynamically (something I tried and it works) we can go from vertical to horizontal and the new gestures will be there. I think we can do:

```js
if (!mVertical) {
  setPageTransformer(null);
  mGestureDetector = null;
  return;
}

Wdyt @devvit ?

Reply:
I seen this
```javascript
// example/ViewPagerExample.js
pageMargin={10}
// Lib does not support dynamically orientation change
orientation="horizontal"
```

I'm not sure if ios switch orientation dynamically,  the scene is rare. Maybe we can add the feature after using ViewPager2.